packages/contracts/contracts/newsroom/Newsroom.sol

```diff
@@ -3,67 +3,199 @@ pragma solidity ^0.4.19;
+  @param contentUri Canonical URI to access the content. The client should then verify that the content has the same hash
+  @param contentHash Keccak256 hash of the content that is linked
+  @return Content ID of the newly published content
 
-    uint id = latestId;
-    latestId++;
+  @dev Emits `ContentPublished` and `RevisionUpdated` events
+  */
+  function publishContent(string contentUri, bytes32 contentHash) public requireRole(ROLE_EDITOR) returns (uint) {
+    uint contentId = contentCount;
+    contentCount++;
 
-    content[id] = Revision(
-      contentHash,
-      contentUri,
-      now
-    );
+    pushRevision(contentId, contentUri, contentHash);
 
-    emit RevisionPublished(msg.sender, id, contentUri);
-    return id;
+    emit ContentPublished(msg.sender, contentId, contentUri);
+    return contentId;
   }
 
-  function publishRevisionSigned(string contentUri, bytes32 contentHash, address author, bytes signature) public requireRole(ROLE_EDITOR) returns (uint) {
-    verifyRevisionSignature(contentHash, author, signature);
+  /**
+  @notice Allows the editor to publish content that has been cryptographically pre-approved by the author.
+  The signature is verified before the content is added to the newsroom.
+  This function requires ROLE_EDITOR or owner to use
+  @param contentUri Canonical URI to access the content. The client should then verify that the content has the same hash
+  @param contentHash Keccak256 hash of the content that is linked
+  @param author The public address of the author that is cryptographically approving this content
+  @param signature The signature of authors approval as per the signing procedure
+  @return Content ID of the newly published content
+
+  @dev See the signing procedure in the dev comments of the whole smart-contract
+  Emits `ContentPublished`, `ContentSigned` and `RevisionUpdated` events
+  */
+  function publishContentSigned(
+    string contentUri,
+    bytes32 contentHash,
+    address author,
+    bytes signature
+  ) external requireRole(ROLE_EDITOR) returns (uint)
+  {
+    uint contentId = publishContent(contentUri, contentHash);
+
+    SignedContent storage signData = signedContent[contentId];
+    signData.author = author;
+
+    pushSignature(signData, contentHash, signature);
+
+    emit ContentSigned(contentId, author);
+  }
 
-    uint id = publishRevision(contentUri, contentHash);
-    signedContent[id] = SignedRevision(
-      author,
-      signature
-    );
+  /**
+  @notice Updates the existing content with a new revision, the content id stays the same while revision id increases afterwards
+  Requires that the content was first published
+  This function can be only called by ROLE_EDITOR or the owner.
+  Can't be used for signed content.
+
+  @dev Emits `RevisionUpdated` event
+  */
+  function updateRevision(uint contentId, string contentUri, bytes32 contentHash) external requireRole(ROLE_EDITOR) {
+    require(!isSigned(contentId));
```

Comment:
editors need to be able to publish unsigned revisions on top of signed ones.  There are a bunch of situations where they may not have access to the original author

Reply:
This is the only one that's blocking in my opinion